app/(admin)/admin/notifications/page.tsx

```diff
@@ -1,16 +1,163 @@
 // Copyright (c) Gridiron Survivor.
 // Licensed under the MIT License.
 
-import { JSX } from 'react';
+'use client';
+import { Button } from '@/components/Button/Button';
+import { getCurrentLeague } from '@/api/apiFunctions';
+import { Input } from '@/components/Input/Input';
+import { JSX, useState } from 'react';
+import { LabelText } from '@/components/LabelText/LabelText';
+import {
+  RadioGroupDefault,
+  RadioGroupDefaultItem,
+} from '@/components/RadioGroupDefault/RadioGroupDefault';
+import { sendEmailNotifications } from './serverFunctions/serverFunctionHelper';
+import { Textarea } from '@/components/Textarea/Textarea';
+import React, { useEffect } from 'react';
+import { getUserTargets } from './serverFunctions/serverFunctionHelper';
 
 /**
  * The admin home page.
  * @returns The rendered AdminHome page.
  */
 const AdminNotifications = (): JSX.Element => {
+  const [content, setContent] = useState<string>('');
+  const [emailSubjects, setEmailSubjects] = useState<string>('all users');
+  const [groupUsers, setGroupUsers] = useState<string[]>([]);
+  const [leagueName, setLeagueName] = useState<string>('');
+  const [subject, setSubject] = useState<string>('');
+
+  /**
+   * To grab all users from the league.
+   * @returns The league data.
+   */
+  const getLeagueData = async (): Promise<void> => {
+    try {
+      const leagueId = '66f1a8e300102bff03ff'; // EMAIL TEST LEAGUE (DO NOT CHANGE)
+      const leagueData = await getCurrentLeague(leagueId);
+      const userTargets = await getUserTargets({
+        userIDs: leagueData.participants,
+      });
+      setGroupUsers(userTargets);
+      setLeagueName(leagueData.leagueName);
+    } catch (error) {
+      throw error;
+    }
+  };
+
+  /**
+   * Handle form submission
+   * @param event - Prevents the default reloading.
+   */
+  const handleSubmit = async (event: React.FormEvent): Promise<void> => {
+    event.preventDefault();
+    await sendEmailNotifications({
+      content,
+      groupUsers,
+      subject,
+    });
+  };
+
+  /**
+   * Function to handle radio selection logic.
+   * @param value - Value of the radio buttons.
+   */
+  const handleRadioChange = (value: string): void => {
+    setGroupUsers([value]);
+    setEmailSubjects(value);
+  };
+
+  useEffect(() => {
+    /**
+     * Fetches the league data.
+     * @returns The league data.
+     */
+    const fetchData = async (): Promise<void> => {
```

Comment:
Does this need to be in a useEfect or can it just be called when the component mounts?